package.json

```diff
@@ -22,53 +25,26 @@
-    "build_closure_core": "node ./tools/make-closure-core.js",
-    "build_global": "npm-run-all clean_dist_global build_esm5_for_rollup && mkdirp ./dist/global && node ./tools/make-umd-bundle.js && npm-run-all build_closure_core clean_dist_esm5_for_rollup",
-    "build_umd": "npm-run-all clean_dist_global && mkdirp ./dist/global && node ./tools/make-umd-bundle.js && npm-run-all build_closure_core",
-    "build_perf": "webdriver-manager update && npm-run-all build_cjs build_global perf",
-    "build_spec": "npm-run-all build_cjs generate_packages",
-    "build_spec_full": "npm-run-all build_spec",
     "build:spec:browser": "echo \"Browser test is not working currently\" && exit -1 && webpack --config spec/support/webpack.mocha.config.js",
-    "clean_dist": "shx rm -rf ./dist",
-    "clean_dist_cjs": "shx rm -rf ./dist/cjs",
-    "clean_dist_esm5": "shx rm -rf ./dist/esm5",
-    "clean_dist_esm5_for_rollup": "shx rm -rf ./dist/esm5_for_rollup",
-    "clean_dist_esm2015": "shx rm -rf ./dist/esm2015",
-    "clean_dist_global": "shx rm -rf ./dist/global",
-    "clean_dist_types": "shx rm -rf ./dist/typings",
-    "commit": "git-cz",
-    "compile_dist_cjs": "tsc -p ./tsconfig/tsconfig.cjs.json",
-    "compile_dist_esm5": "tsc -p ./tsconfig/tsconfig.esm5.json",
-    "compile_dist_esm2015": "tsc -p ./tsconfig/tsconfig.esm2015.json",
-    "compile_dist_esm5_for_rollup": "tsc -p ./tsconfig/tsconfig.esm5.rollup.json",
-    "compile_dist_types": "tsc -p ./tsconfig/tsconfig.types.json",
-    "copy_sources": "mkdirp dist && shx cp -r ./src/ ./dist/src",
     "decision_tree_widget": "cd doc/decision-tree-widget && npm run build && cd ../..",
-    "generate_packages": "node .make-packages.js",
-    "lint_perf": "eslint perf/",
     "lint_spec": "tslint -c tslint.json \"spec/**/*.ts\"",
     "lint_src": "tslint -c tslint.json \"src/**/*.ts\"",
     "lint": "npm-run-all --parallel lint_*",
-    "dtslint": "npm run build_types && dtslint --localTs ./node_modules/typescript/lib --expectOnly ./spec-dtslint",
-    "perf:micro": "node ./perf/micro/index.js",
-    "prepublish": "shx rm -rf ./typings && npm run build_all",
+    "dtslint": "npm run compile && dtslint --localTs ./node_modules/typescript/lib --expectOnly ./spec-dtslint",
+    "prepublishOnly": "npm run build:package",
     "postpublish": "./docs_app/scripts/publish-docs.sh",
     "publish_docs": "./publish_docs.sh",
-    "test": "cross-env TS_NODE_PROJECT=spec/tsconfig.json mocha --opts spec/support/default.opts \"spec/**/*-spec.ts\"",
-    "test_no_cache": "cross-env TS_NODE_PROJECT=spec/tsconfig.json TS_NODE_CACHE=false mocha --opts spec/support/default.opts \"spec/**/*-spec.ts\"",
-    "test_transpile_only": "cross-env TS_NODE_PROJECT=spec/tsconfig.json TS_NODE_TRANSPILE_ONLY=true mocha --opts spec/support/default.opts \"spec/**/*-spec.ts\"",
+    "test": "npm run compile && shx cp -Rf ./dist/cjs ./dist/src && mocha --opts spec/support/default.opts \"dist/spec/**/*-spec.js\"",
```

Comment:
spec have imports to src (../src/**), instead of trying to resolve it in complex way simply just copy cjs output to src allow spec resolves it.

most notably, now mocha runs via transpiled results so no more cache related problem should occur + runs faster as compilation only runs against changed files instead of running all files using ts-node.